From e581b423e7e0df1dc1615d7b35ac17475b2c51d5 Mon Sep 17 00:00:00 2001
From: Alexander Pyhalov <apyhalov@gmail.com>
Date: Wed, 30 Nov 2016 16:09:17 +0300
Subject: [PATCH] Revert "Allow installing a gem inside a rescue block"

This reverts commit ae4b5336efe559b83d7ccd06ffca46f70c6e948e.

This commit breaks documentation generation when install-dir is used

https://github.com/rubygems/rubygems/issues/1470
---
 lib/rubygems/request_set.rb | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/rubygems/request_set.rb b/lib/rubygems/request_set.rb
index 95a8eed..fa5ab1f 100644
--- a/lib/rubygems/request_set.rb
+++ b/lib/rubygems/request_set.rb
@@ -187,6 +187,9 @@ class Gem::RequestSet
       requests << spec
     end
 
+    requests
+  ensure
+    raise if $!
     return requests if options[:gemdeps]
 
     specs = requests.map do |request|
@@ -205,8 +208,6 @@ class Gem::RequestSet
     Gem.done_installing_hooks.each do |hook|
       hook.call inst, specs
     end unless Gem.done_installing_hooks.empty?
-
-    requests
   end
 
   ##
-- 
2.9.2

